src/js/server/devpmt-datasources.js

```diff
@@ -0,0 +1,96 @@
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+var fs = require("fs");
+var JSON5 = require("json5");
+
+gpii.devpmt.loadNPSet = function (prefsetDir, prefsetName, ontologyHandler) {
+    var elod = JSON5.parse(fs.readFileSync(
+        __dirname + prefsetDir + prefsetName + ".json"));
+    var npset = ontologyHandler.rawPrefsToOntology(elod, "flat");
+    return npset;
+};
+
+gpii.devpmt.loadNPSetDocs = function (prefsetDir, prefsetName) {
+    var docs = "";
+    var filename = __dirname + prefsetDir + prefsetName + ".md";
+    if (fs.existsSync(filename)) {
+        docs = fs.readFileSync(filename, {encoding: "UTF8"});
+    }
+    return docs;
+};
+
+/**
+ * If the NP Set does not exist yet, creates a blank one,
+ * otherwise does nothing.
+ */
+gpii.devpmt.addNPSet = function (prefsetDir, npsetName) {
+    var filename = prefsetDir + npsetName + ".json";
+    if (fs.existsSync(filename)) {
+        return;
+    }
+
+    var data = {
+        "contexts": {
+            "gpii-default": {
+                "name": "Default preferences",
+                "preferences": {}
+            }
+        }
+    };
+    gpii.devpmt.saveNPSet(prefsetDir, npsetName, JSON.stringify(data, null, 4));
+};
+
+gpii.devpmt.saveNPSet = function (prefsetDir, npsetName, data) {
+    var filename = prefsetDir + npsetName + ".json";
+    console.log("About to writte to: ", filename);
```

Comment:
The moving finger writte, and having writte, moves on

Reply:
Fixed spelling, updated to fluid.log and added method doc